### PR TITLE
Add support for Cosmos SDK - Address

### PR DESCRIPTION
**Pull request by johnnynanjiang**
- [x] Add TWCoinTypeCosmos
- [x] Add Test for Cosmos address derivation
- [x] Add generated JNI and Swift files
- [x] Add XCTestCase for Cosmos address derivation

### CODE REVIEW

File: src/Coin.cpp [hunk 1]
```diff
@@ -99,9 +100,13 @@ bool TW::validateAddress(TWCoinType coin, const std::string& string) {
     case TWCoinTypeBitcoinCash:
         return Bitcoin::CashAddress::isValid(string) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixBitcoin, TWP2SHPrefixBitcoin});
 
+    case TWCoinTypeCosmos:
+        return Tendermint::Address::isValid(string, HRP_COSMOS) || Tendermint::Address::isValid(string, HRP_COSMOS);
```
Comment (hewigovens) [inline src/Coin.cpp]: what's the point of validate twice with same func?

File: src/Coin.cpp [hunk 2]
```diff
@@ -289,6 +297,15 @@ DerivationPath TW::derivationPath(TWCoinType coin) {
             DerivationPathIndex(0, true),
         };
 
+    case TWCoinTypeCosmos:
```
Comment (hewigovens) [inline src/Coin.cpp]: This is same as previous `DerivationPath(purpose(coin), coin, 0, 0, 0);`
Can you cross checking with ledger to make sure the derivation path is same?

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 3]
```diff
@@ -163,6 +168,7 @@ const char *explorerURLForCoinType(enum TWCoinType type) {
     case TWCoinTypeNimiq: return "https://nimiq.watch";
     case TWCoinTypeStellar: return "https://stellarscan.io";
     case TWCoinTypeAion: return "https://mainnet.aion.network";
+    case TWCoinTypeCosmos: return "http://rpc.hub.certus.one:26657";
```
Comment (hewigovens) [inline src/interface/TWCoinTypeConfiguration.cpp]: This one is not even `https`

File: swift/Tests/CoinAddressDerivationTests.swift [hunk 4]
```diff
@@ -21,6 +21,8 @@ class CoinAddressDerivationTests: XCTestCase {
                     XCTAssertEqual("0xa0629f34c9ea4757ad0b275628d4d02e3db6c9009ba2ceeba76a5b55fb2ca42e", address)
                 case .binance:
                     XCTAssertEqual("tbnb12vtaxl9952zm6rwf7v8jerq74pvaf77fkw9xhl", address)
+                case .cosmos:
+                    XCTAssertEqual("cosmos142j9u5eaduzd7faumygud6ruhdwme98qsy2ekn", address)
```
Comment (vikmeup) [inline swift/Tests/CoinAddressDerivationTests.swift]: derivation is done here now, make sure you double check this on ledger

File: swift/Sources/Blockchains/Cosmos.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import Foundation
+
+public class Cosmos: Blockchain {
+
+    public override init(purpose: Purpose = .bip44) {
+        super.init(purpose: purpose)
+    }
+
+    public var hrp: HRP {
```
Comment (vikmeup) [inline swift/Sources/Blockchains/Cosmos.swift]: or just `public let hrp = HRP.cosmos`

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 6]
```diff
@@ -228,6 +235,7 @@ TWString *_Nonnull TWCoinTypeConfigurationGetName(enum TWCoinType type) {
     case TWCoinTypeNimiq: string = "Nimiq"; break;
     case TWCoinTypeStellar: string = "Stellar"; break;
     case TWCoinTypeAion: string = "Aion"; break;
+    case TWCoinTypeCosmos: string = "Atom"; break;
```
Comment (vikmeup) [inline src/interface/TWCoinTypeConfiguration.cpp]: `Cosmos`

File: src/interface/TWCoinTypeConfiguration.cpp [hunk 7]
```diff
@@ -195,6 +201,7 @@ TWString *_Nonnull TWCoinTypeConfigurationGetID(enum TWCoinType type) {
     case TWCoinTypeNimiq: string = "nimiq"; break;
     case TWCoinTypeStellar: string = "stellar"; break;
     case TWCoinTypeAion: string = "aion"; break;
+    case TWCoinTypeCosmos: string = "atom"; break;
```
Comment (vikmeup) [inline src/interface/TWCoinTypeConfiguration.cpp]: `cosmos`

File: swift/Tests/HDWalletTests.swift [hunk 8]
```diff
@@ -265,4 +265,14 @@ class HDWalletTests: XCTestCase {
         XCTAssertEqual(address.description, "bc1qearv5ezm3xfgy2t98denkzxwst4f35fvz608wa")
         XCTAssertEqual(address.description, address1?.description)
     }
+
+    func testDeriveCosmosAddress() {
+        let wallet = HDWallet(mnemonic: "attract term foster morning tail foam excite copper disease measure cheese camera rug enroll cause flip sword waste try local purchase between idea thank", passphrase: "")
+
+        let key = wallet.getKeyForCoin(coin: .cosmos)
+        let address = CoinType.cosmos.deriveAddress(privateKey: key)
```
Comment (vikmeup) [inline swift/Tests/HDWalletTests.swift]: you don't need this if you did test in `CoinAddressDerivationTests.swift`

File: swift/Tests/Blockchains/CosmosTests.swift [hunk 9]
```diff
@@ -0,0 +1,12 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import XCTest
+import TrustWalletCore
+
+class CosmosTests: XCTestCase {
```
Comment (vikmeup) [inline swift/Tests/Blockchains/CosmosTests.swift]: delete this file?

File: src/Coin.cpp [hunk 10]
```diff
@@ -103,9 +104,13 @@ bool TW::validateAddress(TWCoinType coin, const std::string& string) {
     case TWCoinTypeBitcoinCash:
         return Bitcoin::CashAddress::isValid(string) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixBitcoin, TWP2SHPrefixBitcoin});
 
+    case TWCoinTypeCosmos:
+        return Tendermint::Address::isValid(string, HRP_COSMOS);
+
     case TWCoinTypeDash:
         return Bitcoin::Address::isValid(string, {TWP2PKHPrefixDash, TWP2SHPrefixDash});
```
Comment (vikmeup) [inline src/Coin.cpp]: extra space

File: src/Coin.cpp [hunk 11]
```diff
@@ -297,6 +305,15 @@ DerivationPath TW::derivationPath(TWCoinType coin) {
             DerivationPathIndex(0, true),
         };
 
+    case TWCoinTypeCosmos:
+        return DerivationPath{
+            DerivationPathIndex(purpose(coin), true),
+            DerivationPathIndex(coin, true),
```
Comment (vikmeup) [inline src/Coin.cpp]: @johnnynanjiang did you verify this with ledger? 

I would expect it should use the same format as binance or ethereum

File: src/Coin.cpp [hunk 12]
```diff
@@ -297,6 +305,15 @@ DerivationPath TW::derivationPath(TWCoinType coin) {
             DerivationPathIndex(0, true),
         };
 
+    case TWCoinTypeCosmos:
```
Comment (hewigovens) [inline src/Coin.cpp]: This is same as TWCoinTypeEthereum, could you please revert this change?

File: src/Coin.cpp (hunk 11 above)
Comment (hewigovens) [reply src/Coin.cpp]: Yes, @johnnynanjiang did

File: tests/interface/TWCoinTypeConfigTests.cpp [hunk 13]
```diff
@@ -173,8 +177,8 @@ TEST(TWCoinTypeConfiguration, TWCoinTypeConfigurationGetTransactionURL) {
     auto value24 = WRAPS(TWCoinTypeConfigurationGetTransactionURL(TWCoinTypeAion, txId));
     assertStringsEqual(value24, "https://mainnet.aion.network/#/transaction/123");
 
-    auto value25 = WRAPS(TWCoinTypeConfigurationGetTransactionURL(TWCoinTypeKIN, txId));
-    assertStringsEqual(value25, "https://kinexplorer.com/tx/123");
```
Comment (hewigovens) [inline tests/interface/TWCoinTypeConfigTests.cpp]: one more fix, Did you remove Kin?
